src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/mappers/TransitLayerMapper.java

```diff
@@ -90,29 +95,39 @@ private TransitLayer map() {
             newTripPatternForOld
         );
 
-        for (ServiceDate serviceDate : graph.index.getServiceCodesRunningForDate().keySet()) {
-            // Create LocalDate equivalent to the OTP/GTFS ServiceDate object, serving as the key of
-            // the return Map.
-            LocalDate localDate = ServiceCalendarMapper.localDateFromServiceDate(serviceDate);
-
-            // Create a List to hold the values for one entry in the return Map.
-            List<TripPatternForDate> values = new ArrayList<>();
-
-            // This nested loop could be quite inefficient.
-            // Maybe determine in advance which patterns are running on each service and day.
-            for (org.opentripplanner.model.TripPattern oldTripPattern : allTripPatterns) {
-                TripPatternForDate tripPatternForDate = tripPatternForDateMapper.map(oldTripPattern.scheduledTimetable, serviceDate);
-                if (tripPatternForDate != null) {
-                    values.add(tripPatternForDate);
+        Set<ServiceDate> allServiceDates = graph.index.getServiceCodesRunningForDate().keySet();
+
+        allServiceDates
```

Comment:
Mak a comment here so it is REALY easy to see this this is running in multiple threads.